delivery_postlogistics/models/stock_picking.py

```diff
@@ -0,0 +1,286 @@
+# Copyright 2013-2016 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import base64
+from operator import attrgetter
+
+from odoo import _, api, exceptions, fields, models
+
+from ..postlogistics.web_service import PostlogisticsWebService
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    delivery_type = fields.Selection(related="carrier_id.delivery_type")
+    # TODO: Change the name of this field
+    postlogistics_option_ids = fields.Many2many(
+        comodel_name="postlogistics.delivery.carrier.option", string="Options"
+    )
+
+    delivery_fixed_date = fields.Date(
+        "Fixed delivery date", help="Specific delivery date (ZAW3217)"
+    )
+    delivery_place = fields.Char(
+        "Delivery Place", help="For Deposit item service (ZAW3219)"
+    )
+    delivery_phone = fields.Char(
+        "Phone", help="For notify delivery by telephone (ZAW3213)"
+    )
+    delivery_mobile = fields.Char(
+        "Mobile", help="For notify delivery by telephone (ZAW3213)"
+    )
+
+    @api.onchange("carrier_id")
+    def onchange_carrier_id(self):
+        """ Inherit this method in your module """
+        if not self.carrier_id:
+            return
+        # This can look useless as the field carrier_code and
+        # carrier_type are related field. But it's needed to fill
+        # this field for using this fields in the view. Indeed the
+        # module that depend of delivery base can hide some field
+        # depending of the type or the code
+        carrier = self.carrier_id
+        self.update({"delivery_type": carrier.delivery_type})
+        default_options = carrier.default_options()
+        self.postlogistics_option_ids = default_options
+        result = {
+            "domain": {
+                "postlogistics_option_ids": [
+                    ("id", "in", carrier.postlogistics_available_option_ids.ids)
+                ]
+            }
+        }
+        return result
+
+    @api.onchange("postlogistics_option_ids")
+    def onchange_postlogistics_option_ids(self):
+        if not self.carrier_id:
+            return
+        carrier = self.carrier_id
+        for available_option in carrier.postlogistics_available_option_ids:
+            if (
+                available_option.mandatory
+                and available_option not in self.postlogistics_option_ids
+            ):
+                # XXX the client does not allow to modify the field that
+                # triggered the onchange:
+                # https://github.com/odoo/odoo/issues/2693#issuecomment-56825399
```

Comment:
Not sure this is still valid and nor useful to keep

Reply:
this method will be removed when we remove the field on picking which will be replaced by the product packages assigned to the picking